src/metabase/integrations/slack.clj

```diff
@@ -318,27 +318,83 @@
         channel-id' (get name->id channel-id channel-id)]
     channel-id'))
 
+(defn- poll
+  "Returns true if `(thunk)` returns a result that satisfies the `done?` predicate within the timeout,
+   and false otherwise."
+  [{:keys [thunk done? timeout-ms interval-ms]}]
+  (let [start-time (System/currentTimeMillis)]
+    (loop []
+      (let [response (thunk)]
+        (if (done? response)
+          true
```

Comment:
I think for this case it's fine, but I'd rather have `response` or `nil` returned here, it seems like a generally more useful abstraction then.